### DISCUSSION TITLE
How do you specify command codes for MIDs in the scheduler?

### DISCUSSION THREAD

**Discussion by PavloGV** [jphickey/cfe-eds-framework#9] [Q&A]
How do you specify command codes for MIDs in the scheduler?

**Comment by dmccomas** [comment]
You're going to need a more advanced scheduler than SCH_LAB.

The NASA [scheduler app](https://github.com/nasa/SCH) uses two tables: A message table that defines the messages that can be sent on the software bus. These messages can include data beyond the CCSDS primary header and a scheduler table that defines when to send a particular message from the message table.

[cFS Basecamp](https://github.com/cfs-tools/cfs-basecamp) which uses this toolchain has a similar design in its [kit_sch](https://github.com/cfs-tools/cfs-basecamp/tree/main/apps/kit_sch). KIT_SCH uses the Basecamp app framework design so it uses JSON files for tables. 

- Dave
